### PR TITLE
[finishes #187355105] ft-update-user-password

### PR DESCRIPTION
**Pull request by rwemakevin**
### What does this PR do?
-----------------------------------------------------------------------------------------------------------------------------------------

This PR allows Signed in users to change their password

### Description of Task to be completed?
-----------------------------------------------------------------------------------------------------------------------------------------

The task is about changing user's password. Once a user has access to their account, i.e are logged in, they can update their password in case they want to.

### How should this be manually tested?
-----------------------------------------------------------------------------------------------------------------------------------------

1.Run `npm install` to install all packages

2.Set the node environments inside .env as shown in the table below

| variable | value |
|--------|--------|
| PORT | value |
| DATABASE_URL| value |
| NODE_ENV | value |
| JWT_SECRET| value | 

3.Run `npm run build` to transpile ts file

4.Finally run `npm run start` to start the development server

5.then test the functionality using swaggerUI or postman
   swaggerUI :localhost:PORT/docs
   endpoint to test: api/user/changepassword
   
  ### NOTE
  -----------------------------------------------------------------------------------------------------------------------------------------
  You need to log into your acount in order to update your password

### CODE REVIEW

File: src/tests/updateUserPassword.test.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+jest.mock('jsonwebtoken', () => ({
+  verify: jest.fn(),
+}));
+
+// // Mocking JWT
+// jest.mock('../helps/jwt', () => ({
+//     JWT: {
+//         generateJwt: jest.fn(),
+//     },
+// }));
```
Comment (UmuhireJessie) [inline src/tests/updateUserPassword.test.ts]: Remove these!

File: src/tests/updateUserPassword.test.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    jest.clearAllMocks();
+  });
+  it('should extract token, password, newPassword, and verifyNewPassword from request body', async () => {
+    await UserController.updatePassword(req as Request, res as Response);
+
+    // Assert that the controller extracts the correct values from the request body
+    //expect(req.token).toBe('mockedToken');
+    //expect(req.body.password).toBe('oldPassword');
+    // expect(req.body.newPassword).toBe('newPassword');
+    // expect(req.body.verifyNewPassword).toBe('newPassword');
```
Comment (UmuhireJessie) [inline src/tests/updateUserPassword.test.ts]: Remove these comments

File: src/controllers/userControllers.ts [hunk 3]
```diff
@@ -280,6 +280,67 @@ export default class UserController {
...
   }
+
+  static async updatePassword(req: any, res: Response) {
+    const { token } = req;
+    const { password, newPassword, verifyNewPassword } = req.body;
+
+    try {
+      const getDecodedToken = jwt.verify(token, secret);
+      const userId = getDecodedToken.userId;
+      console.log(getDecodedToken);
```
Comment (UmuhireJessie) [inline src/controllers/userControllers.ts]: Remove this console

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Good work @rwemakevin, please work on the comments above!

File: src/tests/updateUserPassword.test.ts (hunk 1 above)
Comment (rwemakevin) [reply src/tests/updateUserPassword.test.ts]: This is fixed now

File: src/tests/updateUserPassword.test.ts (hunk 2 above)
Comment (rwemakevin) [reply src/tests/updateUserPassword.test.ts]: This is fixed too

**Review by UmuhireJessie — APPROVED** [review]

**Review by lilika67 — APPROVED** [review]

**Review by jniyonkuru — APPROVED** [review]

**Review by venuste-20002 — APPROVED** [review]

**Pull request merged**